core/domain/user_query_domain.py

```diff
@@ -25,21 +25,32 @@
 from core import utils
 from core.constants import constants
 
+from typing import List, TypeVar, Type, Optional
+from datetime import datetime
+
+T = TypeVar('T', bound='UserQuery')
+
 attribute_names = [ # pylint: disable=invalid-name
     predicate['backend_attr'] for predicate in (
         constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
 
 UserQueryParams = collections.namedtuple( # pylint: disable=invalid-name
-    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names))
+    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names)) # type: ignore
```

Comment:
Why is the ignore needed?

Reply:
Line 37 and 38 were throwing the same error of namedTuple expects a list or tuple , so added the ignore but same error

Reply:
@vojtechjelinek You can read about this issue [here](https://github.com/python/mypy/issues/848) on github and [here](https://stackoverflow.com/questions/44638360/mypy-error-list-or-tuple-literal-expected-as-the-second-argument-to-namedtuple) on StackOverflow. I think using `type: ignore[misc]` with an explanatory comment will be the way out. What do you think? Thanks!

Reply:
Ah, I see. Yeah I'm fine with using `# type: ignore[misc]` but there should be a comment explaining why it is used.

Reply:
Done